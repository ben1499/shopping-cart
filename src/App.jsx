import { useState } from 'react'
// import './App.css'
import Header from './components/Header';
import Home from './components/Home';
import Footer from './components/Footer';
import Router from './Router';

function App() {

  return (
    <>
      <Header />
      <Router />
      <Footer />
    </>
  )
}

export default App
