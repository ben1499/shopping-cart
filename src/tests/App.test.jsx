
import { render, screen } from "@testing-library/react"
import App from "../App"
import { expect, vi } from "vitest";
import { act } from "react-dom/test-utils";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import vi-fetch-mock

describe("App component", () => {
    let mockProducts = [
        {
            "id": 1,
            "title": "Blue shirt",
            "price": 109.95,
        }
    ]

    global.fetch = vi.fn(() => Promise.resolve({
        json: () => Promise.resolve(mockProducts)
    }));


    it("renders App component", () => {
        const { container } = render(<BrowserRouter>
            <App />
        </BrowserRouter>);

        expect(container).toMatchSnapshot();
    })

    it("renders fetched data", async () => {
        act(() => {
            render(<BrowserRouter>
                <App />
            </BrowserRouter>);
        })

        const user = new userEvent.setup();

        const link = screen.getByRole("link", { name: "Shop" })

        await user.click(link);
        expect(screen.getByText(/Blue shirt/i)).toBeInTheDocument();
    })
})