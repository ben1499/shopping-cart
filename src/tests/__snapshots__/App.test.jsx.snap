// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders App component 1`] = `
<div>
  <header>
    <h1>
      <a
        class="sc-guDLey bysVtm"
        href="/"
      >
        Shopsy
      </a>
    </h1>
    <nav>
      <ul>
        <li>
          <a
            class="sc-guDLey bysVtm"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="sc-guDLey bysVtm"
            href="/shop"
          >
            Shop
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <footer
    class="sc-dmyCSP edkHKz"
  >
    <p>
      Â© 2024 Shopsy
    </p>
  </footer>
</div>
`;
